src/OrbitBase/TemporaryFile.cpp

```diff
@@ -36,17 +39,13 @@ ErrorMessageOr<void> TemporaryFile::Init() {
 
   fd_ = unique_fd(fd);
 #elif defined(_WIN32)
-  // _mktemp_s requires string to be null-terminated.
-  file_path += '\0';
-  errno_t errnum = _mktemp_s(file_path.data(), file_path.size());
+  // _mktemp_s expects the `size` to include the NULL character at the end.
+  errno_t errnum = _mktemp_s(file_path.data(), file_path.size() + 1);
   if (errnum != 0) {
     return ErrorMessage{
         absl::StrFormat("Unable to create a temporary file: %s", SafeStrerror(errnum))};
   }
 
-  // get rid of '\0' at the end
-  file_path = std::string(file_path.c_str());
-
```

Comment:
Note that `std::string` already guarantees that its buffer is null terminated (since C++11). So there is no need to add an extra null characters, hence I removed that.